### PR TITLE
Add progress bar in incomplete properties

### PR DESCRIPTION
**Pull request by Zefling**
To make the progression more visual. 

Screenshot_20210228_132224

### CODE REVIEW

**Comment by LeaVerou** [comment]
I do not understand what this progress bar is showing. Also, it's extremely subtle (which may or may not be ok depending on what it's depicting).

**Comment by SebastianZ** [comment]
@LeaVerou If I understand the change correctly, the progress bar is just another visual representation for how much of a feature is covered by a browser besides the smileys and the colors.

I just checked out the branch locally and I can see the progress bars, though I think they could be made a little more visible.

Sebastian

File: tests.js [hunk 1]
```diff
@@ -4327,7 +4327,7 @@ window.Specs = {
 					"tr": "#rubypos",
 					"dev": "#rubypos"
 				},
-				"tests": ["over", "under", "inter-character"]
```
Comment (SebastianZ) [inline tests.js:4330]: It looks like that change shouldn't be part of this patch.

Sebastian

Comment (Zefling) [reply tests.js:4330]: oups...

**Comment by LeaVerou** [comment]
I wonder if we should make the progress bar more prominent (by not having it as a background, but as a line at the top of each element) and remove the smiley faces, which are harder to quantify.

**Comment by Zefling** [comment]
Like this? 
image

**Comment by LeaVerou** [comment]
Yes, except I think it would look better on top instead of bottom (also try increasing the padding a bit on the side with the progress bar)

**Comment by Zefling** [comment]
If anyone has better to suggest.
image

**Comment by LeaVerou** [comment]
I still think a thin progress bar on top would be better than a background one, and would allow stronger contrast too.

File: style.css [hunk 2]
```diff
@@ -311,6 +319,27 @@ details li[class]:after {
...
+
+.almost-pass .progress-bar { background: #6da300;  }
+
+.slightly-buggy .progress-bar { background: #969600;  }
+
+.buggy .progress-bar { background:  #be6d00}
+
+.very-buggy .progress-bar { background: #be3600; }
+
+.fail .progress-bar { background: #be1b00;; }
```
Comment (SebastianZ) [inline style.css:339]: Here's a redundant semicolon.

File: style.css [hunk 3]
```diff
@@ -311,6 +319,27 @@ details li[class]:after {
 
 .very-buggy:after { content: ':o'; }
 
+.progress-bar {
+    height: 100%;
```
Comment (SebastianZ) [inline style.css:323]: You should use tabs for indentation here.

**Review by SebastianZ — CHANGES_REQUESTED** [review]
The remarks are just little things.

I really like this feature, though I agree with @LeaVerou that the full-height progress bars are too much and prefer the thin version you've shown before. I don't have a preference whether it is at the top or bottom, though.

File: style.css (hunk 2 above)
Comment (Zefling) [reply style.css]: Fixed

File: style.css (hunk 3 above)
Comment (Zefling) [reply style.css]: Fixed

**Comment by Zefling** [comment]
Update for this: 
    
image

**Comment by LeaVerou** [comment]
LGTM visually.
I think the code would be quite simplified if instead of using an extra element, you used a background image. You can pass in the progress via a CSS variable, then the rest can be in CSS.

You can do something like `linear-gradient(rgb(0 0 0 / .3), rgb(0 0 0 / .3)) no-repeat 0 0 / calc(var(--progress) * 1%) .2em)` for the effect I see above.

**Comment by Zefling** [comment]
Isn't it a problem if it doesn't work on old browsers?

**Comment by Zefling** [comment]
With CSS var : 
image

**Comment by LeaVerou** [comment]
> Isn't it a problem if it doesn't work on old browsers?

No, since this is an additional way to communicate the result, not something essential. As long as nothing *breaks* in older browsers, we're good. 

Also note that CSS variables are supported in pretty much every browser in use today except IE11 and Opera Mini. We're talking 95% of Web users. https://caniuse.com/css-variables

**Comment by Zefling** [comment]
Ok, It's done. ;)

**Review by LeaVerou — APPROVED** [review]

**Comment by SebastianZ** [comment]
Three notes:
1. Firefox obviously has rounding issues for the width of the progress bar. Sometimes it's 7px wide, sometimes 6px. This causes the bar to look a little bit odd. Maybe just use `px` instead?
Progress bar display in Firefox

2. Chrome and Firefox differ regarding the width of the progress bar. In Chrome it's always 5px for me, in Firefox it's one or two pixels wider. @LeaVerou This looks like a bug in Chrome to me. `font-size` has a computed value of 16px in both browsers for me. And the `0.4em` for `background-size` correctly result in `6.4px` as computed value in Firefox, though it is `4.8px` in Chrome. Any idea why? That issue aside, the bar looks a little too wide for my taste. It should rather be `4px` or max. `5px`, also to avoid the issue mentioned above.

3. You may want to increase the top padding a little (maybe by `0.1em` or `0.2em`) like Lea mentioned earlier. This will make the text look vertically centered again (except for cases with 0% implementation).

Sebastian

**Comment by Zefling** [comment]
I changed to .2em

It's true, the padding is not good in Chromium: 
image

**Comment by SebastianZ** [comment]
If I choose a padding of `0.6em` instead of `0.7em`, it seems fine to me in Chrome and Firefox.

Progress bar and paddings with `padding-top: 0.6em`

I still get the rounding errors in Firefox (compare the width of the bar for `border-top-right-radius` and `border-bottom-right-radius`) but I think it's not a show-stopper.

Sebastian

**Comment by Zefling** [comment]
Is it better?

**Comment by SebastianZ** [comment]
Yes, works for me. Thank you!

Sebastian

**Review by SebastianZ — APPROVED** [review]
